src/test/utils/debounce.test.ts

```diff
@@ -3,10 +3,114 @@
  *  Licensed under the MIT License. See LICENSE in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { debounce } from '../../service/utils/debounce';
+
+// This delay is used as the delay given to `debounce`, but also assertions are made that each test should take at least (delay - 1) milliseconds and at most 2 * delay milliseconds
+// This could result in some test fragility; if so a longer delay should reduce the frequency of that

```

Comment:
The lower bound is OK, but the upper bound, in my experience, is just an invitation to have regular, annoying false positive failures. The cloud VMs running tests are prone to very long pauses (10s of seconds) at random times.

If you really want to ensure that the upper bound is holding in a reasonable way, run the test 100 times or so, drop the 10 lowest and 10 highest results and then check the median of the remaining results.

Reply:
Good point. I added an assertion failure in the first test to be extra emphatic that _only_ the second debounce call should get executed (the value assertion for `x` only ensures that one of them gets executed, not particularly which one). I removed the upper bound on time.